### DISCUSSION TITLE
system-bridge as systemd user service

### DISCUSSION THREAD

**Discussion by zhulik** [timmo001/system-bridge#1114] [Help] [answered]
Hello, 

Is there a way to run headless system bridge as a systemd user service that uses nodejs from OS repositories? I'm no nodejs developer, but I believe there must be a way to compile the whole headless app into one js file that can be executed with something like `/usr/bin/node system-bridge.js config.json`

**Comment by timmo001** [comment]
This is probably possible. There should be an environment variable to set, to make the app run in CLI mode (no GUI features) for `SB_CLI=true` and `SB_GUI=false`.

The Docker image has made a start with this in https://github.com/timmo001/system-bridge/blob/master/rootfs/etc/services.d/app/run

**Reply by zhulik** [reply]
This is almost what I was looking for, but I'd like to build a version without bundled node and huge binaries, just pure js executed in system nodejs

**Comment by timmo001** [comment, marked as answer]
Documentation for 3.x.x here contains a systemd script:

https://system-bridge.timmo.dev/docs/running

**Reply by Tovrin** [reply]
I'm a bit of a noob to linux so bear with me.

The first line says "sudo cp .scripts/linux/system-bridge.service /etc/systemd/system/", but where is the .scripts directory? I'm at a bit of a loss.

**Reply by bwduncan** [reply]
Click on "Code" at the top of this repository. You will find the `.scripts` directory.
